discovery/parse.go

```diff
@@ -27,6 +27,19 @@ func NewApp(name string, labels map[string]string) (*App, error) {
 	}, nil
 }
 
+func ExtractPort(name string) (nameWithoutPort string, port string) {
```

Comment:
This function only extracts the port and takes it out of the name, that's why it's not called from `NewAppFromString`, the port is not part of an App. We could assign the port to an App flag though.